modules/core/src/test/java/org/apache/ignite/internal/processors/security/cache/closure/ContinuousQueryWithTransformerRemoteSecurityContextCheckTest.java

```diff
@@ -0,0 +1,136 @@
+import org.apache.ignite.cache.query.ScanQuery;
+import org.apache.ignite.lang.IgniteClosure;
+import org.apache.ignite.lang.IgniteRunnable;
+import org.junit.Test;
+
+/**
+ * Testing operation security context when the ScanQuery, RemoteFilterFactory or TransformerFactory of a
+ * ContinuousQueryWithTransformer is executed on remote node.
+ * <p>
+ * The initiator node broadcasts a task to 'run' node that starts a ContinuousQueryWithTransformer's component. That
+ * component is executed on 'check' node. On every step, it is performed verification that operation securitycontext is
+ * the initiator context.
+ */
+public class ContinuousQueryWithTransformerRemoteSecurityContextCheckTest extends
+    AbstractContinuousQueryRemoteSecurityContextCheckTest {
+    /** Test InitialQuery of ContinuousQueryWithTransformer. */
+    @Test
+    public void testInitialQuery() {
+        Consumer<ContinuousQueryWithTransformer<Integer, Integer, Integer>> consumer =
+            new Consumer<ContinuousQueryWithTransformer<Integer, Integer, Integer>>() {
+                @Override public void accept(ContinuousQueryWithTransformer<Integer, Integer, Integer> q) {
+                    q.setInitialQuery(new ScanQuery<>(INITIAL_QUERY_FILTER));
+                    q.setRemoteTransformerFactory(new TestTransformerFactory(false));
+                }
+            };
+
+        runAndCheck(grid(SRV_INITIATOR), operation(consumer));
+        runAndCheck(grid(CLNT_INITIATOR), operation(consumer));
+    }
+
+    /** Test RemoteFilterFactory of ContinuousQueryWithTransformer. */
+    @Test
+    public void testRemoteFilterFactory() {
+        Consumer<ContinuousQueryWithTransformer<Integer, Integer, Integer>> consumer =
+            new Consumer<ContinuousQueryWithTransformer<Integer, Integer, Integer>>() {
+                @Override public void accept(ContinuousQueryWithTransformer<Integer, Integer, Integer> q) {
+                    q.setRemoteFilterFactory(new TestRemoteFilterFactory());
+                    q.setRemoteTransformerFactory(new TestTransformerFactory(false));
+                }
+            };
+
+        runAndCheck(grid(SRV_INITIATOR), operation(consumer));
+        runAndCheck(grid(CLNT_INITIATOR), operation(consumer));
+    }
+
+    /** Test TransformerFactory of ContinuousQueryWithTransformer. */
+    @Test
+    public void testTransformerFactory() {
+        Consumer<ContinuousQueryWithTransformer<Integer, Integer, Integer>> consumer =
+            new Consumer<ContinuousQueryWithTransformer<Integer, Integer, Integer>>() {
+                @Override public void accept(ContinuousQueryWithTransformer<Integer, Integer, Integer> q) {
+                    q.setRemoteTransformerFactory(new TestTransformerFactory(true));
+                }
+            };
+
+        runAndCheck(grid(SRV_INITIATOR), operation(consumer));
+        runAndCheck(grid(CLNT_INITIATOR), operation(consumer));
+    }
+
+    /** Test operation. */
```

Comment:
Convert to multi-line javadoc according to coding guidelines, add @param and @return tags.